openreview/conference/builder.py

```diff
@@ -0,0 +1,349 @@
+from . import webfield
+from . import invitation
+
+class ConferenceType(object):
+
+    @classmethod
+    def homepage_webfield_template(cls):
+        return "noBlindConferenceWebfield.js"
+
+    @classmethod
+    def submission_reply(cls, id, subject_areas, additional_fields):
+        return {}
+
+class SingleBlindConferenceType(ConferenceType):
+
+    @classmethod
+    def homepage_webfield_template(cls):
+        return "noBlindConferenceWebfield.js"
+
+class DoubleBlindConferenceType(ConferenceType):
+
+    @classmethod
+    def homepage_webfield_template(cls):
+        return "homepage.js"
+
+    @classmethod
+    def submission_reply(cls, id, subject_areas, additional_fields):
+
+        content = {
+            'title': {
+                'description': 'Title of paper.',
+                'order': 1,
+                'value-regex': '.{1,250}',
+                'required':True
+            },
+            'authors': {
+                'description': 'Comma separated list of author names. Please provide real names; identities will be anonymized.',
+                'order': 2,
+                'values-regex': "[^;,\\n]+(,[^,\\n]+)*",
+                'required':True
+            },
+            'authorids': {
+                'description': 'Comma separated list of author email addresses, lowercased, in the same order as above. For authors with existing OpenReview accounts, please make sure that the provided email address(es) match those listed in the author\'s profile. Please provide real emails; identities will be anonymized.',
+                'order': 3,
+                'values-regex': "([a-z0-9_\-\.]{2,}@[a-z0-9_\-\.]{2,}\.[a-z]{2,},){0,}([a-z0-9_\-\.]{2,}@[a-z0-9_\-\.]{2,}\.[a-z]{2,})",
+                'required':True
+            },
+            'keywords': {
+                'description': 'Comma separated list of keywords.',
+                'order': 6,
+                'values-regex': "(^$)|[^;,\\n]+(,[^,\\n]+)*"
+            },
+            'TL;DR': {
+                'description': '\"Too Long; Didn\'t Read\": a short sentence describing your paper',
+                'order': 7,
+                'value-regex': '[^\\n]{0,250}',
+                'required':False
+            },
+            'abstract': {
+                'description': 'Abstract of paper.',
+                'order': 8,
+                'value-regex': '[\\S\\s]{1,5000}',
+                'required':True
+            },
+            'pdf': {
+                'description': 'Upload a PDF file that ends with .pdf',
+                'order': 9,
+                'value-regex': 'upload',
+                'required':True
+            }
+        }
+
+        if subject_areas:
+            content['subject areas'] = {
+                'order' : 5,
+                'description' : "Select or type subject area",
+                'values-dropdown': subject_areas,
+                'required': True
+            }
+
+        order = 10
+        for field in additional_fields:
```

Comment:
I'm not certain about this, but:

It might be nice if the structure of the objects in `additional_fields` was the same as the structure of the objects in "content". That might also open up the possibility of inserting additional fields elsewhere in the list, rather than only allowing them to appear at the end.

But again, I'm not certain if this is a good idea

Reply:
Yes, I thought about that. I like to iterate a list instead of an object because with the list you keep the order. Other option is to have the structure of the content and validate the order. It can be collisions between them and that requires to change all the content. Let me think about it.